synapse/storage/databases/main/roommember.py

```diff
@@ -534,6 +534,32 @@ async def get_local_users_in_room(self, room_id: str) -> List[str]:
             desc="get_local_users_in_room",
         )
 
+    async def check_local_user_in_room(self, user_id: str, room_id: str) -> bool:
```

Comment:
This is similar to `check_user_in_room` that is in `synapse/api/auth.py` but that one has some specific logic and can check for `leave` so I decided to leave it as-is.

https://github.com/matrix-org/synapse/blob/a25a37002c851ef419d12925a11dd8bf2233470e/synapse/api/auth.py#L69-L119